gnome-initial-setup/pages/site/gis-site-search-entry.c

```diff
@@ -0,0 +1,518 @@
+    {
+      if (strncmp (p, word, word_len) != 0)
+        continue;
+
+      if (p > (char *)full_name)
+        {
+          char *prev = g_utf8_prev_char (p);
+
+          /* Make sure p points to the start of a word */
+          if (g_unichar_isalpha (g_utf8_get_char (prev)))
+            continue;
+
+          /* If we're matching the first word of the key, it has to
+           * match the first word of the field.
+           * Eg, it either matches the start of the string
+           * (which we already know it doesn't at this point) or
+           * it is preceded by the string ", " or "(" (which isn't actually
+           * a perfect test.)
+           */
+          if (is_first_word)
+            {
+              if (prev == (char *)full_name ||
+                  ((prev - 1 <= full_name && strncmp (prev - 1, ", ", 2) != 0)
+                    && *prev != '('))
+                continue;
+            }
+        }
+
+      if (whole_word && g_unichar_isalpha (g_utf8_get_char (p + word_len)))
+        continue;
+
+      return p;
+    }
+
+  return NULL;
+}
+
+static gboolean
+match_compare_name (const char *key, const char *name)
+{
+  gboolean is_first_word = TRUE;
+  size_t len;
+
+  /* Ignore whitespace before the string */
+  key += strspn (key, " ");
+
+  /* All but the last word in KEY must match a full word from NAME,
+   * in order (but possibly skipping some words from NAME).
+   */
+  len = strcspn (key, " ");
+  while (key[len])
+    {
+      name = find_word (name, key, len, TRUE, is_first_word);
+      if (!name)
+        return FALSE;
+
+      key += len;
+      while (*key && !g_unichar_isalnum (g_utf8_get_char (key)))
+        key = g_utf8_next_char (key);
+      while (*name && !g_unichar_isalnum (g_utf8_get_char (name)))
+        name = g_utf8_next_char (name);
+
+      len = strcspn (key, " ");
+      is_first_word = FALSE;
+    }
+
+  /* The last word in KEY must match a prefix of a following word in NAME */
+  if (len == 0)
+    return TRUE;
+
+  /* if we get here, key[len] == 0 */
+  g_assert (len == strlen (key));
+
+  return find_word (name, key, len, FALSE, is_first_word) != NULL;
+}
+
+static gboolean
+matcher (GtkEntryCompletion *completion, const char *key,
+         GtkTreeIter *iter, gpointer user_data)
+{
+  char *compare_name;
+  gboolean match;
+
+  gtk_tree_model_get (gtk_entry_completion_get_model (completion), iter,
+                      GIS_SITE_SEARCH_ENTRY_COL_LOCAL_COMPARE_NAME, &compare_name,
+                      -1);
+
+  match = match_compare_name (key, compare_name);
+
+  g_free (compare_name);
+
+  return match;
+}
+
+static gboolean
+match_selected (GtkEntryCompletion *completion,
+                GtkTreeModel       *model,
+                GtkTreeIter        *iter,
+                gpointer            entry)
+{
+  GisSiteSearchEntryPrivate *priv =
+      gis_site_search_entry_get_instance_private (entry);
+  GisSite *site = NULL;
+
+  if (model != priv->model)
```

Comment:
I don't think this code can be reached. The documentation of the `match-selected` signal mentions that
```
Note that model is the model that was passed to gtk_entry_completion_set_model().
```

and we ever only call `gtk_entry_completion_set_model()` with `priv->model`...

Reply:
Actually, it turns out we never actually hit this callback anyhow, so I'm just removing it.